src/rgw/rgw_op.cc

```diff
@@ -110,23 +112,47 @@ static int parse_range(const char *range, off_t& ofs, off_t& end, bool *partial_
     goto done;
 
   *partial_content = true;
+  end = -2;
```

Comment:
why? doesn't it overrides the end calculations that is used for the out of range error